src/triggers.js

```diff
@@ -724,15 +724,18 @@ async function builtInTriggerValidator(options, request, auth) {
         if (opt.required) {
           requiredParam(key);
         }
-        if (opt.type) {
-          const type = getType(opt.type);
-          const valType = Array.isArray(val) ? 'array' : typeof val;
-          if (valType !== type) {
-            throw `Validation failed. Invalid type for ${key}. Expected: ${type}`;
+        const optional = opt.type && opt.required === false && !opt.default && val === undefined;
```

Comment:
I believe it should be `const optional = !opt.required && val === undefined;`

- We need to remove opt.type because we want options to be validated even if there is no specified type
- `!opt.required` might worked better because default is not required, right?
- `!opt.default` is not needed here since default value has been already applied to val in the beginning of the function